dtfs-central-api/src/v1/controllers/portal/deal/update-deal.controller.js

```diff
@@ -196,26 +202,36 @@ exports.removeFacilityIdFromDeal = removeFacilityIdFromDeal;
 
 exports.updateDealPut = async (req, res) => {
   try {
-    if (!ObjectId.isValid(req.params.id)) {
+    const {
+      params: { id: dealId },
+      body: { user, dealUpdate, auditDetails },
+    } = req.body;
+
+    if (!ObjectId.isValid(dealId)) {
       return res.status(400).send({ status: 400, message: 'Invalid Deal Id' });
     }
 
-    const dealId = req.params.id;
-    const { user, dealUpdate } = req.body;
+    validateAuditDetails(auditDetails);
 
     // TODO: Refactor callback with status check
     return findOneDeal(dealId, async (deal) => {
-      if (deal) {
-        const response = await updateDeal(dealId, dealUpdate, user, deal, req.routePath);
-        const status = isNumber(response?.status, 3);
-        const code = status ? response.status : 200;
-
-        return res.status(code).json(response);
+      if (!deal) {
+        return res.status(404).send({ status: 404, message: 'Deal not found' });
       }
+      const response = await updateDeal(dealId, dealUpdate, user, deal, req.routePath, auditDetails);
+      const status = isNumber(response?.status, 3);
```

Comment:
Legacy/beyond scope but why is `isNumber` needed?

Reply:
Oh no this is because it handles validation in `updateDeal` and sometimes returns `{ status: 400, message: 'whatever' }`